### ISSUE TITLE
RuntimeError: values[i]->type()->isSubtypeOf(value_type) INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/ir/ir.cpp":1650, please report a bug to PyTorch.

### ISSUE THREAD

**Issue by johnson-magic** [pytorch/pytorch#81197]
### 1. by running the code below, you can reproduce the error
```
import torch
@torch.jit.script
def rotate_points_export(points):
    return points

def xxx(input):
    outputs = {}
    outputs['a'] = input
    outputs['b'] = input

    outputs['b'] = rotate_points_export(outputs['b'])
    return outputs

points = torch.rand((1,2,3,4))

model = torch.jit.trace(xxx, points, strict=False)
torch.jit.save(model, 'xx.pt')

```
### 2. Error messages shows:
```
    model = torch.jit.trace(xxx, points, strict=False)
  File "/home/intsig/codes/depedencies/hhdetection/lib/python3.8/site-packages/torch/jit/_trace.py", line 778, in trace
    traced = torch._C._create_function_from_trace(
RuntimeError: values[i]->type()->isSubtypeOf(value_type) INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/ir/ir.cpp":1650, please report a bug to PyTorch. 
```

### Versions

```
Collecting environment information...
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.13.0-52-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.1.74
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1080 Ti
Nvidia driver version: 470.129.06
cuDNN version: Probably one of the following:
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.1
[pip3] torch==1.7.1
[pip3] torchvision==0.8.2
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.2.0           h06a4308_296  
[conda] mkl-service               2.3.0            py38h27cfd23_1  
[conda] mkl_fft                   1.3.0            py38h42c9631_2  
[conda] mkl_random                1.2.1            py38ha9443f7_2  
[conda] numpy                     1.20.1           py38h93e21f0_0  
[conda] numpy-base                1.20.1           py38h7d8b39e_0  
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1  
```

**Comment by davidberard98** [comment]
I can't repro on master branch, so I'm assuming that this has been fixed since pytorch 1.7.1. If you can still repro on newer versions then please reopen the issue with more details.

**Comment by johnson-magic** [comment]
Thank you for your answer. Yes, you are right. When I use newer pytorch version(i.e. pytorch1.9.0), this error disappear. @davidberard98
